Spigot-API-Patches/0283-Add-ElderGuardianAppearanceEvent.patch

```diff
@@ -0,0 +1,81 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 19 Mar 2021 23:39:21 -0400
+Subject: [PATCH] Add ElderGuardianAppearanceEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/ElderGuardianAppearanceEvent.java b/src/main/java/io/papermc/paper/event/entity/ElderGuardianAppearanceEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1de578530ac90b91ccd7064006fc7aff2127f12b
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/ElderGuardianAppearanceEvent.java
+@@ -0,0 +1,69 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.ElderGuardian;
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++
++import java.util.Collections;
++import java.util.List;
++
++/**
++ * Is called when an {@link org.bukkit.entity.ElderGuardian} appears in front of a {@link org.bukkit.entity.Player}.
++ */
++public class ElderGuardianAppearanceEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
++
++    private boolean cancelled;
++    private final Player affectedPlayer;
++
++    public ElderGuardianAppearanceEvent(@NotNull Entity what, @NotNull Player affectedPlayer) {
++        super(what);
++        this.affectedPlayer = affectedPlayer;
++    }
++
++    /**
++     * Get the player affected by the guardian appearance.
```

Comment:
Maybe it would be good to also add them being the player triggering the effect?

Reply:
It isn't caused by a specific player, simply it scans for nearby players on the tick method.
![image](https://user-images.githubusercontent.com/23108066/120115563-412ed780-c152-11eb-93ba-6516029b8181.png)

Reply:
TIL! Thanks!
Looks good for merge, then :)